### ISSUE TITLE
Why throw errGasUintOverflow when memory size overflows?

### ISSUE THREAD

**Issue by viniul** [ethereum/go-ethereum#16698]
In line 178 in `core/vm/interpreter.go`, an errGasUintOverflow is returned. However, it seems to that this error is returned when actually the memory size overflows. Is this true? And if so, is this not a bit misleading? 

Best, 
Vincent
https://github.com/ethereum/go-ethereum/blob/45bd4feddeadfbde5d1e560797155aacb0abbadf/core/vm/interpreter.go#L178

**Comment by karalabe** [comment]
The EVM in theory is considered to have infinite amount of memory. In practice, the memory is always limited by the available gas, which is many many orders of magnitude smaller than what your machine could support. If your memory request overflows the uint64 counter, then it surely overflows the gas allowance too.

TL;DR You cannot overflow memory by design, only gas.
